govcd/edgegateway.go

```diff
@@ -131,11 +132,81 @@ func (eGW *EdgeGateway) AddDhcpPool(network *types.OrgVDCNetwork, dhcppool []int
 
 }
 
+// Temporary fix for existing Remove functions - until ID support
+func (eGW *EdgeGateway) RemoveNATMappingRule(networkHref, natType, externalIP, internalIP, port string) (Task, error) {
+	return eGW.RemoveNATPortRule(networkHref, natType, externalIP, port, internalIP, port)
+}
+
+// Temporary fix for existing Remove functions - until ID support
+func (eGW *EdgeGateway) RemoveNATPortRule(networkHref, natType, externalIP, externalPort, internalIP, internalPort string) (Task, error) {
+	newEdgeConfig := eGW.EdgeGateway.Configuration.EdgeGatewayServiceConfiguration
+
+	// Take care of the NAT service
+	newNatService := &types.NatService{}
+
+	newNatService.IsEnabled = newEdgeConfig.NatService.IsEnabled
+	newNatService.NatType = newEdgeConfig.NatService.NatType
+	newNatService.Policy = newEdgeConfig.NatService.Policy
+	newNatService.ExternalIP = newEdgeConfig.NatService.ExternalIP
+
+	for _, natRule := range newEdgeConfig.NatService.NatRule {
+
+		// Kludgy IF to avoid deleting DNAT rules not created by us.
+		// If matches, let's skip it and continue the loop
+		interfaceId, err := extractObjectIDfromPath(natRule.GatewayNatRule.Interface.HREF)
+		networkHrefId, err := extractObjectIDfromPath(networkHref)
+		if err != nil {
+			return Task{}, nil
+		}
+		if natRule.RuleType == natType &&
+			natRule.GatewayNatRule.OriginalIP == externalIP &&
+			natRule.GatewayNatRule.OriginalPort == externalPort &&
+			interfaceId == networkHrefId {
+			util.Logger.Printf("[DEBUG] REMOVING %s Rule: %#v", natRule.RuleType, natRule.GatewayNatRule)
+			continue
+		}
+		util.Logger.Printf("[DEBUG] KEEPING %s Rule: %#v", natRule.RuleType, natRule.GatewayNatRule)
+		newNatService.NatRule = append(newNatService.NatRule, natRule)
+	}
+
+	newEdgeConfig.NatService = newNatService
+
+	newRules := &types.EdgeGatewayServiceConfiguration{
+		Xmlns:      types.XMLNamespaceVCloud,
+		NatService: newNatService,
+	}
+
+	apiEndpoint, _ := url.ParseRequestURI(eGW.EdgeGateway.HREF)
+	apiEndpoint.Path += "/action/configureServices"
+
+	// Return the task
+	return eGW.client.ExecuteTaskRequest(apiEndpoint.String(), http.MethodPost,
+		"application/vnd.vmware.admin.edgeGatewayServiceConfiguration+xml", "error reconfiguring Edge Gateway: %s", newRules)
+
+}
+
+func extractObjectIDfromPath(locationPath string) (string, error) {
```

Comment:
What is the purpose of this function? Depending on the kind of path and the kind of ID that we want to extract, this function would look differently. 
Could you add a unit test to ensure that this does what we expect?

Reply:
Add comment. P.s. with next PR all these function will be removed

Reply:
A general reminder about function naming: we agreed to use `Id` instead of `ID` (and `From` instead of `from`).

Reply:
Tx. Won't change as it will be removed